tests/unit/test-firefox/test.firefox.js

```diff
@@ -511,40 +512,240 @@ describe('firefox', () => {
+           /The request "profileName" profile name cannot be resolved/
+         );
+       }
+    );
+
     it('resolves to a FirefoxProfile instance', () => withBaseProfile(
-      (baseProfile) => {
+      async (baseProfile) => {
         const configureThisProfile = (profile) => Promise.resolve(profile);
-        return firefox.useProfile(baseProfile.path(), {configureThisProfile})
-          .then((profile) => {
-            assert.instanceOf(profile, FirefoxProfile);
-          });
+        const createProfileFinder = () => {
+          return (profilePath) => Promise.resolve(profilePath);
+        };
+        const profile = await firefox.useProfile(baseProfile.path(), {
+          configureThisProfile,
+          createProfileFinder,
+        });
+        assert.instanceOf(profile, FirefoxProfile);
+      }
+    ));
+
+    it('looks for profile path if passed a name', () => withBaseProfile(
+      async (baseProfile) => {
+        const fakeGetProfilePath = sinon.spy(() => baseProfile.path());
+        const createProfileFinder = () => {
+          return fakeGetProfilePath;
+        };
+        const isFirefoxDefaultProfile = sinon.spy(
+          () => Promise.resolve(false)
+        );
+        await firefox.useProfile('profileName', {
+          createProfileFinder,
+          isFirefoxDefaultProfile,
+        });
+        sinon.assert.calledOnce(fakeGetProfilePath);
+        sinon.assert.calledWith(
+          fakeGetProfilePath,
+          sinon.match('profileName')
+        );
+      }
+    ));
+
+    it('checks if named profile is default', () => withBaseProfile(
+      async (baseProfile) => {
+        const createProfileFinder = () => {
+          return () => Promise.resolve(baseProfile.path());
+        };
+        const isFirefoxDefaultProfile = sinon.spy(
+          () => Promise.resolve(false)
+        );
+        await firefox.useProfile('profileName', {
+          createProfileFinder,
+          isFirefoxDefaultProfile,
+        });
+        sinon.assert.calledOnce(isFirefoxDefaultProfile);
+        sinon.assert.calledWith(
+          isFirefoxDefaultProfile,
+          sinon.match('profileName')
+        );
       }
     ));
 
     it('configures a profile', () => withBaseProfile(
       (baseProfile) => {
         const configureThisProfile =
           sinon.spy((profile) => Promise.resolve(profile));
-        const app = 'fennec';
         const profilePath = baseProfile.path();
-        return firefox.useProfile(profilePath, {app, configureThisProfile})
+        return firefox.useProfile(profilePath, {configureThisProfile})
           .then((profile) => {
             sinon.assert.called(configureThisProfile);
             sinon.assert.calledWith(configureThisProfile, profile);
-            assert.equal(configureThisProfile.firstCall.args[1].app, app);
           });
       }
     ));
 
   });
 
+  describe('defaultCreateProfileFinder', () => {
+
+    function prepareReaderTest(readProfileReturns) {
```

Comment:
Nit, how about renaming this to `prepareProfileFinderTest`?

also it think that it may be easier to guess what the parameters of this test helper do if we pass them as object properties, and it would also allow us to expand the helper for the test that needs it (e.g. we may add a `getPathResult` to inject an expected result into the out test spy and complete the test "returns a finder that resolves a profile name" as suggested below), I mean something like:

```js
function prepareReaderTest({readProfileResult, getPathResult}) {
  ...
}

it('returns a finder that resolves a profile name', async () => {
      const {
        ...
      } = prepareReaderTest({
       readProfileResult: ...,
       getPathResult: ...,
      });
      
      ...
```

Reply:
These changes have been applied in the follow up commit 7d8b703.